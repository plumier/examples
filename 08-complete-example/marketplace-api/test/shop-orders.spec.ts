import supertest from "supertest"

import createApp from "../src/app"
import { closeConnection, createCart, ignore } from "./_helper"


const oIgnore = { ...ignore, date: expect.any(String), user: { ...ignore }, address: { ...ignore } }

afterEach(async () => {
    await closeConnection()
})

describe("Shop Orders", () => {
    it("Should able to list orders", async () => {
        const app = await createApp({ mode: "production" })
        const { user, shops: [putra] } = await createCart(app)
        await supertest(app.callback())
            .post("/api/carts/checkout")
            .set("Authorization", `Bearer ${user.token}`)
            .expect(200)
        const { body: orders } = await supertest(app.callback())
            .get(`/api/shops/${putra.shop.id}/orders`)
            .set("Authorization", `Bearer ${putra.staffs[0].token}`)
            .expect(200)
        expect(orders[0]).toMatchSnapshot(oIgnore)
    })
    it("Should accessible by owner", async () => {
        const app = await createApp({ mode: "production" })
        const { user, shops: [putra] } = await createCart(app)
        await supertest(app.callback())
            .post("/api/carts/checkout")
            .set("Authorization", `Bearer ${user.token}`)
            .expect(200)
        await supertest(app.callback())
            .get(`/api/shops/${putra.shop.id}/orders`)
            .set("Authorization", `Bearer ${putra.owner.token}`)
            .expect(200)
    })
    it("Should not accessible by non shop user", async () => {
        const app = await createApp({ mode: "production" })
        const { user, shops: [putra, ali] } = await createCart(app)
        await supertest(app.callback())
            .post("/api/carts/checkout")
            .set("Authorization", `Bearer ${user.token}`)
            .expect(200)
        await supertest(app.callback())
            .get(`/api/shops/${putra.shop.id}/orders`)
            .set("Authorization", `Bearer ${ali.owner.token}`)
            .expect(403)
    })
})