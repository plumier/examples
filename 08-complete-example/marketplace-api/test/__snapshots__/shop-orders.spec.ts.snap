// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop Orders Should able to list orders 1`] = `
Object {
  "address": Object {
    "address": "Br. Guntur Kemenuh Blahbatuh Gianyar",
    "city": "Gianyar",
    "country": "Indonesia",
    "createdAt": Any<String>,
    "deleted": false,
    "id": Any<Number>,
    "updatedAt": Any<String>,
    "zipCode": "80281",
  },
  "createdAt": Any<String>,
  "date": Any<String>,
  "deleted": false,
  "id": Any<Number>,
  "state": "ReceivedBySeller",
  "updatedAt": Any<String>,
  "user": Object {
    "createdAt": Any<String>,
    "deleted": false,
    "id": Any<Number>,
    "name": "Jane Dane",
    "updatedAt": Any<String>,
  },
}
`;
